dialogue-core/src/main/java/com/palantir/dialogue/core/RetryingChannel.java

```diff
@@ -113,10 +114,16 @@ private void logRetry(Throwable throwable) {
 
         private ListenableFuture<Response> wrap(ListenableFuture<Response> input) {
             return Futures.catchingAsync(
-                    Futures.transformAsync(input, this::success, MoreExecutors.directExecutor()),
+                    Futures.transformAsync(wrapWithTracing(input), this::success, MoreExecutors.directExecutor()),
```

Comment:
We don't capture it in our tests, but all our real channels are wrapped with code to produce tracing spans. It doesn't have a retry index, but the order should make it clear.
What if we wrapped the outermost channel returned from Channels.create with another traced channel instead?

Reply:
sure